### PR TITLE
Enrol: Fix typos

### PR DESCRIPTION
**Pull request by eloycoto**
Fix some logs messages.
Fix swagger definition to be correct.
Check ns namespace just if enrol info is empty.

Signed-off-by: Eloy Coto <eloy.coto@acalustra.com>

### CODE REVIEW

**Review by masayag — APPROVED** [review]

**Comment by gabriel-farache** [comment]
What do you think about either changing the parameter "features" in  EdgeDeviceSignedRequestSpec to a list of Feature or renaming it to "feature" as in the CRD it is not a list and that the plural use here is misleading when you create the request

**Comment by eloycoto** [comment]
> What do you think about either changing the parameter "features" in EdgeDeviceSignedRequestSpec to a list of Feature or renaming it to "feature" as in the CRD it is not a list and that the plural use here is misleading when you create the request

At the end, is hardware features, no? so for me are features, but I'll say that @jakub-dzon set the first name on the Edge device spec, so let's Jakub to be involved.

**Review by gabriel-farache — COMMENTED** [review]
When you look at https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/ you see that complex names are either camelCased or hyphen separated, why changing target-namespace with snake_case ?

File: docs/design/http_api_swagger.md [hunk 1]
```diff
@@ -473,7 +473,7 @@ Status: Internal Server Error
 | Name | Type | Go type | Required | Default | Description | Example |
 |------|------|---------|:--------:| ------- |-------------|---------|
 | features | [EnrolmentInfoFeatures](#enrolment-info-features)| `EnrolmentInfoFeatures` |  | |  |  |
-| target-namespace | string| `string` |  | `"default"`|  |  |
+| target_namespace | string| `string` |  | `"default"`|  |  |
```
[the remark this answers is not in the document]
Comment (eloycoto) [reply docs/design/http_api_swagger.md:476]: This is not kubernetes CRD, this is swagger API, and the convention that we use is `_`, this is why I changed.

[the remark this answers is not in the document]
Comment (gabriel-farache) [reply docs/design/http_api_swagger.md:476]: oh yes, indeed, changes only applies to restapi files, the CRD spec is in a separated file indeed, sorry

**Review by gabriel-farache — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**